app/data.py

```diff
@@ -22,51 +24,76 @@ def get_local_movie_df() -> Any:
     df = pd.read_csv("./app/input/movies.csv")
     return df
 
+
+def get_tmdb_id(id: int, df_movies=Depends(get_movie_df)) -> int:
+    movie_row = df_movies.loc[df_movies["movieId"] == id]
+    if movie_row.empty:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND)
+    tmdbId = int(movie_row["tmdbId"])
+    return tmdbId
+
 @lru_cache
 def get_local_movie_recommendations_df() -> Any:
     df = pd.read_csv("./app/input/ratings.csv")
     return df
 
 
 async def get_movie_details_from_tmdb(
-    tmdbId: int, settings: Settings = Depends(get_settings)
+    tmdb_id: int = Depends(get_tmdb_id), settings: Settings = Depends(get_settings)
 ) -> MoviePublic:
-    print(settings)
     async with httpx.AsyncClient() as client:
         response = await client.get(
-            settings.tmdb_base_url + f"/movie/{tmdbId}?api_key={settings.tmdb_api_key}"
+            settings.tmdb_base_url + f"/movie/{tmdb_id}?api_key={settings.tmdb_api_key}"
         )
         response_json = response.json()
         return MoviePublic(
-            id=tmdbId,
+            id=tmdb_id,
             title=response_json["original_title"],
             thumbnail_url=f"{settings.tmdb_images_base_url}{response_json['poster_path']}",
         )
 
-
-async def get_movie_details(id: int, df_movies=Depends(get_movie_df)) -> MoviePublic:
-
-    movie_row = df_movies.loc[df_movies["movieId"] == id]
-    tmdbId = int(movie_row["tmdbId"])
-    movie = await get_movie_details_from_tmdb(tmdbId=tmdbId, settings=get_settings())
-
+async def get_movie_details(
+    id: int,
+    movie: MoviePublic = Depends(get_movie_details_from_tmdb)
+) -> MoviePublic:
     # Replace tmdbId with dataset ID
     movie.id = id
 
     return movie
 
+class MovieDetailFetcher:
+
+    def __init__(self, settings: Settings = Depends(get_settings)):
```

Comment:
Good Job with refactoring. Why are we making __call__ method async to fetch movie_details?

Reply:
It needs to be `async` since we `await` inside:
```python

tmdb_movie_details = await get_movie_details_from_tmdb(tmdb_id, self.settings)

return await get_movie_details(movie_id, tmdb_movie_details)
```

Reply:
Got it, thanks for clarifying